File: frontend/src/services/Post.ts
export interface CompletePost {
  idAuthor: number
  authorName: string
  content: string
  createdAt: Date
  likes: number
  comments: JSON[]
}

// public class CompletePost {
//   public string? AuthorName { get; set; }
//   public int IdAuthor { get; set; }
//   public string ? Content { get; set; }
//   public DateTime ? CreatedAt { get; set; }
//   public int ? Likes { get; set; }
//   public ICollection<Comment> ? Comments { get; set; }
//   public int IdPost { get; set; }
//   public int ? IdForum { get; set; }
//   public string ? ForumName { get; set; }
// }

File: backend/Model/Post.cs
using System;
using System.Collections.Generic;

namespace backend.Model;

public partial class Post
{
    public int Id { get; set; }

    public string? Content { get; set; }

    public DateTime? CreatedAt { get; set; }

    public int? FkUser { get; set; }

    public int? FkForum { get; set; }

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual Forum? FkForumNavigation { get; set; }

    public virtual DataUser? FkUserNavigation { get; set; }

    public virtual ICollection<Like> Likes { get; set; } = new List<Like>();
}
